### ISSUE TITLE
modprobe no such device error after remove/rescan

### ISSUE THREAD

**Issue by drybalka** [Witko/nvidia-xrun#151]
It may be a wrong place to ask this question, but still I could not find the answer anywhere. It seems that after removing the card and then rescanning it again i cannot load nvidia anymore. I get:
`modprobe: ERROR: could not insert 'nvidia': No such device`
The output of dmesg is:
`[  103.648841] nvidia-nvlink: Nvlink Core is being initialized, major device number 234
[  103.648845] NVRM: The NVIDIA GPU 0000:01:00.0
               NVRM: (PCI ID: 10de:1c8c) installed in this system has
               NVRM: fallen off the bus and is not responding to commands.
[  103.652050] nvidia: probe of 0000:01:00.0 failed with error -1
[  103.652060] NVRM: The NVIDIA probe routine failed for 1 device(s).
[  103.652061] NVRM: None of the NVIDIA devices were initialized.
[  103.652437] nvidia-nvlink: Unregistered the Nvlink Core, major device number 234`

I made a dry-run of the script and experimented with executing commands manually. It seems everything working just fine, if only i do not run the remove command. More precisely, I can load the drivers, start xinit with nvidia working correctly, close the X server, unload the drivers and everything seem to be working. I checked that pci buses are correct, `lsmod` shows drivers load and unload completely. I also checked with `lspci` that card disappears correctly after remove and appears again after rescan. However, after remove/rescan procedure i can no longer load nvidia and get the aforementioned errors.

Did you by any chance know how to solve this problem?

**Comment by afro-coder** [comment]
Yeah even I'm seeing this error, no clue as to why this happens

**Comment by afro-coder** [comment]
Okay I did some fidgeting and ran these two commands
`nvidia-modprobe`
`nvidia-persistenced`
And now I can load nvidia back again

**Comment by drybalka** [comment]
Sorry, now I just get "modprobe: FATAL: module nvidia is in use." when trying to unload nvidia, even though nothing except `nvidia-persistenced` is running.

**Comment by afro-coder** [comment]
Yep kill that nvidia-persistenced it'll not let you unload nvidia, I was
trying to find a fix but I'm not getting the time
